iothub_client/tests/iothubtransport_mqtt_common_int_ut/iothubtransport_mqtt_common_int_ut.c

```diff
@@ -0,0 +1,1172 @@
+static const char* TEST_MQTT_IGNORED_TOPICS= "devices/myDeviceId/messages/devicebound/" TEST_TOPICS_TO_IGNORE TEST_TOPICS_TO_NOT_IGNORE;
+
+char* expectedNotIgnoredKeys[] = {"devices", "to", "exp", "on"};
+char* expectedNotIgnoredValues[] = {"valueToApp1", "valueToApp2", "valueToApp3", "valueToApp4" };
+TEST_EXPECTED_APPLICATION_PROPERTIES expectedNotIgnored = { expectedNotIgnoredKeys, expectedNotIgnoredValues, 4};
+TEST_EXPECTED_MESSAGE_PROPERTIES mostlyIgnoredPropertiesExpected = { NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, &expectedNotIgnored};
+
+static const char* mqttNoMatchTopic[] = {
+    "",
+    "ThisIsNotCloseToBeingALegalTopic",
+    "/device/",
+    "devices/",
+    "devices/myDeviceId/messages",
+    "devices/myDeviceId/messages/deviceboun",
+    "/devices/myDeviceId/messages/devicebound",
+    // These are legal topics but as we're not subscribed to them they should be ignored.
+    "$iothub/twin/twinData",
+    "iothub/methods/methodData",
+    "devices/myDeviceId/modules/thisIsModuleID/inputs/" TEST_INPUT_QUEUE_1 "/"
+};
+static const size_t mqttNoMatchTopicLength = sizeof(mqttNoMatchTopic) / sizeof(mqttNoMatchTopic[0]);
+
+static const char* emptyPropertyMQTTTopics[] = {
+    "devices/myDeviceId/messages/devicebound/",
+    "devices/myDeviceId/messages/devicebound/&",
+    "devices/myDeviceId/messages/devicebound/&&",
+    "devices/myDeviceId/messages/devicebound/&&&",
+    "devices/myDeviceId/messages/devicebound/=",
+    "devices/myDeviceId/messages/devicebound/fooBar",
+    //"devices/myDeviceId/messages/devicebound/==",
+};
+
+static const size_t emptyMQTTTopicsLength = sizeof(emptyPropertyMQTTTopics) / sizeof(emptyPropertyMQTTTopics[0]);  
+TEST_EXPECTED_MESSAGE_PROPERTIES noProperties = { NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL};
+
+static const char* TEST_MQTT_MESSAGE_APP_PROPERTIES_1 = "devices/myDeviceId/messages/devicebound/customKey1=customValue1";
+char* expectedKey1[] = {"customKey1"};
+char* expectedValue1[] = {"customValue1"};
+TEST_EXPECTED_APPLICATION_PROPERTIES app1 = { expectedKey1, expectedValue1, 1};
+TEST_EXPECTED_MESSAGE_PROPERTIES expectedAppProperties1 = { NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, &app1};
+
+static const char* TEST_MQTT_MESSAGE_APP_PROPERTIES_2 = "devices/myDeviceId/messages/devicebound/customKey1=customValue1&customKey2=customValue2";
+char* expectedKey2[] = {"customKey1", "customKey2"};
+char* expectedValue2[] = {"customValue1", "customValue2"};
+TEST_EXPECTED_APPLICATION_PROPERTIES app2 = { expectedKey2, expectedValue2, 2};
+TEST_EXPECTED_MESSAGE_PROPERTIES expectedAppProperties2 = { NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, &app2};
+
+#define TEST_APP_PROPERTY_KEY1 "temperature%2FAlert1"
+#define TEST_APP_PROPERTY_KEY2 "temperature%2FAlert2"
+#define TEST_APP_PROPERTY_KEY3 "temperature%2FAlert3"
+
+#define TEST_APP_PROPERTY_VALUE1 "false%251"
+#define TEST_APP_PROPERTY_VALUE2 "false%252"
```

Comment:
should these be 
```suggestion
#define TEST_APP_PROPERTY_VALUE1 "false1%25"
#define TEST_APP_PROPERTY_VALUE2 "false2%25"
```
or was this done on purpose (the next one is this way)

Reply:
fixed.